crates/uv-fs/src/path.rs

```diff
@@ -192,6 +195,52 @@ pub fn normalize_path(path: &Path) -> Result<PathBuf, std::io::Error> {
     Ok(ret)
 }
 
+/// Normalize a path, removing things like `.` and `..`.
+///
+/// Unlike [`normalize_absolute_path`], this works with relative paths and does never error.
+///
+/// Note that we can theoretically go beyond the root dir here (e.g. `/usr/../../foo` becomes
+/// `/../foo`), but that's not a (correctness) problem, we will fail later with a file not found
+/// error with a path computed from the user's input.
+///
+/// # Examples
+///
+/// In: `../../workspace-git-path-dep-test/packages/c/../../packages/d`
+/// Out: `../../workspace-git-path-dep-test/packages/d`
+///
+/// In: `workspace-git-path-dep-test/packages/c/../../packages/d`
+/// Out: `workspace-git-path-dep-test/packages/d`
+///
+/// In: `./a/../../b`
+/// Out: `../b`
+pub fn normalize_path(path: &Path) -> PathBuf {
```

Comment:
This function is more lenient than the other path normalization function. Can/should we merge the two? I don't fully understand the constraints on the other normalization function, so i kept them apart for starters.

Reply:
I don't fully understand the constraints either.

This function reminds me of [`std::path::absolute`](https://doc.rust-lang.org/std/path/fn.absolute.html). Should we use that instead here, or are the critical differences in semantics? (The phrasing in your docs of not touching the file system and possibly returning a path that does not exist were what reminded me of `absolute`.)

I typed all that up, but now I see that this routine is specifically handing relative paths and not making them absolute. So `absolute` is probably a no-go.

Reply:
I found the difference:

> Note that no other normalization takes place; in particular, `a/c`
> and `a/b/../c` are distinct, to account for the possibility that `b`
> is a symbolic link (so its parent isn't `a`).

We assume they are the same and normalize this way.